src/types/page.rs

```diff
@@ -116,6 +173,44 @@ pub enum PageRangesPart {
     Range(Numeric, Numeric),
 }
 
+derive_or_from_str!(@deser_impl PageRangesPart where "a page, a page range, or a separator",
+    fn visit_map<A>(self, map: A) -> Result<Self::Value, A::Error>
+        where A: serde::de::MapAccess<'de>, {
+        use serde::{de, Deserialize};
+
+        #[derive(Deserialize)]
+        #[serde(rename_all = "kebab-case")]
+        enum Inner {
+            Ampersand,
+            Comma,
+            EscapedRange(Numeric, Numeric),
+            SinglePage(Numeric),
+            Range(Numeric, Numeric),
+        }
```

Comment:
Out of curiosity, is this deserializing something like

```jsonc
{
  // PageRanges
  "ranges": [
     // PageRangesPart::Ampersand
     "ampersand",
     // PageRangesPart::Comma
     "comma",
     // PageRangesPart::EscapedRange
     { "escaped-range": [12, 34] }
  ]
}
```
?
Is this used somewhere?
(I'm aware the FromStr impl covers the usual cases)



Looking further around, this seems to be just for consistency with the rest of types, so that seems acceptable.